lite/dbprovider.go

```diff
@@ -0,0 +1,260 @@
+package lite
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	amino "github.com/tendermint/go-amino"
+	cryptoAmino "github.com/tendermint/tendermint/crypto/encoding/amino"
+	dbm "github.com/tendermint/tendermint/libs/db"
+	log "github.com/tendermint/tendermint/libs/log"
+	lerr "github.com/tendermint/tendermint/lite/errors"
+	"github.com/tendermint/tendermint/types"
+)
+
+type DBProvider struct {
+	logger log.Logger
+	label  string
+	db     dbm.DB
+	cdc    *amino.Codec
+	limit  int
+}
+
+func NewDBProvider(label string, db dbm.DB) *DBProvider {
```

Comment:
Shouldn't it return PersistentProvider?

Reply:
It's better to return the concrete type, (whereas for function arguments it's better to accept interfaces when appropriate).

Reply:
We are not very consistent on this. If we don't return interface how do you know (without reading all code) what interface is implemented with a struct?